### DISCUSSION TITLE
Some fields have different types in stravalib.model.SummaryActivity  and stravalib.model.DetailedActivity

### DISCUSSION THREAD

**Discussion by jlelong** [hozn/stravalib#568] [General]
Thanks a lot for library!

Fields related to elevation, duration and speed have a float type in `SummaryActivity` but more precise types in `DetailedActivity` such as `DistanceType`, `ElevationType`, VelocityType`. Could it be possible to use the same types in both `SummaryActivity` and `DetailedActivity` to make sure  a `DetailedActivity` can be cast to a `SummaryActivity`?

I would like to be able to use a `DetailedActivity` when a `SummaryActivity` is expected but I cannot figure out how to do it.

**Comment by enadeau** [comment]
Hi, thanks for the issue. I think you are right some of the value are more specialized only the detailed one but not on the summary activity. I'll open a MR to fix that if you want to have a look and make sure it covers what you noticed it would be great.

**Comment by enadeau** [comment]
I've move this to #570

**Comment by jsamoocha** [comment]
This was a bug. Thanks for reporting!
